### PR TITLE
Adapt to IPV6 envs

### PR DESCRIPTION
**Pull request by raul-arabaolaza**
While running `workflow-durable-task-step` tests on a proprietary environment with only IPV6 networking enabled I found some parts of RJR that were not fully IPV6 compatible. This PR tries to fix those parts so RJR can be used in both IPV4 and IPV6 environments.

### Testing done

```[tasklist]
### Submitter checklist
- [ ] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [ ] Ensure that the pull request title represents the desired changelog entry
- [ ] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java [hunk 1]
```diff
@@ -671,7 +671,7 @@ public URL getUrl() throws MalformedURLException {
         if (port == 0) {
             throw new IllegalStateException("This method must be called after calling #startJenkins.");
         }
-        return new URL("http://" + host + ":" + port + "/jenkins/");
```
Comment (raul-arabaolaza) [inline src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:674]: This failed when `host` was an IPV6 ip

**Review by timja — APPROVED** [review]

**Review by rsandell — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Review by jglick — APPROVED** [review]
Looks right but @PereBueno why did this not come up during testing of #712?

**Comment by jtnord** [comment]
> Looks right but @PereBueno why did this not come up during testing of #712?

Most likely system configuration.

Some hosts/systems/OSes call `::1` `localhost-6` causing a miss-match between the name and the address when used before this change.

Other hosts have 2 entries for `localhost`, one for v4 and one for v6.

File: src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java (hunk 1 above)
Comment (Vlatombe) [reply src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java:674]: I think this line was the only necessary change. The other change caused test regressions in other parts, so I'm proposing a revert of them in https://github.com/jenkinsci/jenkins-test-harness/pull/776

**Pull request merged**